src/main/java/com/cx/restclient/httpClient/CxHttpClient.java

```diff
@@ -493,8 +493,10 @@ public void addCustomHeader(String name, String value) {
         } catch (CxTokenExpiredException ex) {
             if (retry) {
                 logTokenError(httpMethod, statusCode, ex);
-                login(lastLoginSettings);
-                return request(httpMethod, contentType, entity, responseType, expectStatus, failedMsg, isCollection, false);
+                if(lastLoginSettings != null) {
+                    login(lastLoginSettings);
+                    return request(httpMethod, contentType, entity, responseType, expectStatus, failedMsg, isCollection, false);
```

Comment:
What is the treatment in case lastLoginSettings == null?

Reply:
Then we don't retry and just throw an exception.